src/main/ruby/core/truffle/cext.rb

```diff
@@ -1268,6 +1268,22 @@ def rb_hash_set_ifnone(hash, value)
     hash.default = value
   end
 
+  ST_CONTINUE = 0
+  ST_STOP = 1
+  ST_DELETE = 2
+
+  def rb_hash_foreach(hash, func, farg)
+    hash.each do |key, value|
+      st_result = foreign_call_with_block(func, key, value, farg)
+
+      case st_result
+      when ST_CONTINUE then next
```

Comment:
This `next` is not really needed here, and incurs the cost of an exception in the interpreter.

Reply:
the case (the `when`) can stay there, just no need for `next` imho.